### PR TITLE
Demonstrating you can use readonly record struct

### PR DESCRIPTION
**Pull request by JeffAshton**
This isn't meant to be merged.  Just proving you can benefit from C# 10.0 features and still support old target frameworks.

### CODE REVIEW

File: src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <OutputType>Library</OutputType>
     <PackageId>LaunchDarkly.ServerSdk</PackageId>
     <RootNamespace>LaunchDarkly.Sdk.Server</RootNamespace>
-    <LangVersion>7.3</LangVersion>
+    <LangVersion>10.0</LangVersion>
```
Comment (JeffAshton) [inline src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj:16]: The C# Language is not the same as the TargetFramework.  You can `record struct` on the following target frameworks:

* net20
* net35
* net451
* net452
* net46
* net461
* net462
* net47
* net471
* net472
* net48
* netcoreapp1.0
* netcoreapp1.1
* netcoreapp2.0
* netcoreapp2.1
* netcoreapp2.2
* netcoreapp3.0
* netcoreapp3.1
* netstandard1.0
* netstandard1.1
* netstandard1.2
* netstandard1.3
* netstandard1.4
* netstandard1.5
* netstandard1.6
* netstandard2.0
* netstandard2.1
* net5.0
* net6.0

File: src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj [hunk 2]
```diff
@@ -38,6 +38,10 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="IsExternalInit" Version="1.0.2">
```
Comment (JeffAshton) [inline src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj:41]: This package is compile time only.  Injects the falling class into your build:

[IsExternalInit](https://source.dot.net/#System.Private.CoreLib/IsExternalInit.cs,7c39b2fb54deae81):
```cs
using System.ComponentModel;

namespace System.Runtime.CompilerServices
{
    [EditorBrowsable(EditorBrowsableState.Never)]
    internal static class IsExternalInit {
    }
}
```

But it's a package dependency:

image

File: src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj (hunk 1 above)
Comment (JeffAshton) [reply src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj:16]: Output from `net20`:

image

**Comment by eli-darkly** [comment]
@JeffAshton Thanks for the suggestion, although this would be more useful as proof of concept if the CI passed (which would require changing the build platform to .NET 6.0, since .NET 5.0 cannot compile C# language version 10).

**Comment by JeffAshton** [comment]
> @JeffAshton Thanks for the suggestion, although this would be more useful as proof of concept if the CI passed (which would require changing the build platform to .NET 6.0, since .NET 5.0 cannot compile C# language version 10).

Yes, CI would need to be updated too.  But if you pull the code down and compile you'll see there is no issue.

**Comment by eli-darkly** [comment]
@JeffAshton To be clear, there is no issue _as long as you're building the SDK in .NET 6.0_. The SDK would be compatible with older target frameworks at link time (i.e. when using it as a NuGet package) and at runtime, but the source would not be buildable if you were using the .NET build tools for older platforms. In the past it's been the case that people building on older platforms could consume our library as source if they wanted to— not only as a compiled package; it's not a common use case but we have always supported it before, in all of our SDKs for compiled languages.

**Pull request closed without merging**